src/tree/gpu_hist/row_partitioner.cuh

```diff
@@ -46,8 +46,18 @@ class RowPartitioner {
    */
   /*! \brief Range of row index for each node, pointers into ridx below. */
   std::vector<Segment> ridx_segments_;
-  dh::TemporaryArray<RowIndexT> ridx_a_;
-  dh::TemporaryArray<bst_node_t> position_a_;
+  dh::caching_device_vector<RowIndexT> ridx_a_;
```

Comment:
It is preferable to use `dh::TemporaryArray` over `dh::caching_device_vector` because of thrust synchronisation in recent versions.

Reply:
So keep the double buffering, but do it using `dh::TemporaryArray`